### ISSUE TITLE
Example: Add more usage of derived data

### ISSUE THREAD

**Issue by brandonroberts** [ngrx/platform#1199]
## I would be willing to submit a PR for the docs :heart:

[ ] Yes (Assistance is provided if you need help submitting a pull request)
[ ] No

Derived data already exists in the Store, so we should add more examples of it. One example is being logged in based on the authenticated user instead of the `loggedIn` property in the store.

**Comment by timdeschryver** [comment]
@kouMatsumoto yep, exactly!

**Comment by timdeschryver** [comment]
Is the above enough for you @mgred, to get started? 😄

**Comment by mgred** [comment]
Got it @timdeschryver!
Thanks for the example @kouMatsumoto, I will have a look and keep you all updated here.

**Comment by etiennecrb** [comment]
Hello!
About this, what is the NGRX team best practice for selectors on large store? I have a main reducer whose `ActionReducerMap` has 16 keys (so 16 features), if I put all my selectors in the main `reducers.ts` file, it would have more than 1000 LOC. Currently, I have a `my-feature.selectors.ts` file in each that import the substate selector from `reducers.ts`. But I find it a bit weird...

Example usage of my current solution:
* store/
  * index.ts
  * books/
    * books.actions.ts
    * books.effects.ts
    * books.reducer.ts
    * books.selectors.ts

> books/books.selectors.ts

```typescript
import { createSelector } from '@ngrx/store';

import { selectBooksState } from '../index';
import * as fromBooks from './books.reducer';

export const getSelectedBookId = createSelector(
  selectBooksState,
  fromBooks.getSelectedBookId
);
```

**Comment by mgred** [comment]
Hey guys,

I started the work on this and begun with the given example.
There is already a PR (#1286) that tracks everything.

In `projects/example-app/src/app/auth/reducers/index.ts` the linter reported of incompatible types in line 20:

```typescript
export const reducers: ActionReducerMap<AuthState> = {
  status: fromAuth.reducer,
  loginPage: fromLoginPage.reducer,
};
```
I then added the `AuthActionUnion` type as the second type parameter to the definition:

```typescript
export const reducers: ActionReducerMap<AuthState, AuthActionUnion> = {
  status: fromAuth.reducer,
  loginPage: fromLoginPage.reducer,
};
```

Even though the linter reports an error the compilation and serving of the app worked nevertheless.

Was this left off on purpose? And should that stay off?

There are some other files reporting the same _Error_.
